src/hotspot/share/opto/loopopts.cpp

```diff
@@ -791,6 +791,11 @@ Node *PhaseIdealLoop::conditional_move( Node *region ) {
     // Ignore Template Assertion Predicates with OpaqueTemplateAssertionPredicate nodes.
     return nullptr;
   }
+  if (bol->is_OpaqueMultiversioning() && bol->as_OpaqueMultiversioning()->is_useless()) {
```

Comment:
Isn't any `OpaqueMultiversioning` we find here supposed to be useless? Can we assert that instead?

Reply:
That is essencially what we I already do. If we find a **useful** node, then we just hit the assert further down.

But if you want then I can refactor it explicitly.

Reply:
It might be more explicit with a separate assert that an `OpaqueMultiversioning` should be useless. And we can avoid the `is_useless()` check in product - but of course that should not make much of a difference performance wise :-)